index.js

```diff
@@ -21,18 +23,15 @@ module.exports = class ServeDrive extends ReadyResource {
     this.connections = new Set()
     this.server = opts.server || http.createServer()
     this.server.on('request', this._onrequest.bind(this))
-
-    this._onfilter = opts.filter
-  }
-
-  async _open () {
-    await Promise.resolve() // Wait a tick, so you don't rely on server.address() being sync sometimes
```

Comment:
Port is available:
```js
const serve = new ServeDrive(...)
console.log(serve.address().port) // => 7000
```

Port is not available:
```js
const serve = new ServeDrive(...)
console.log(serve.address().port) // Throws: Cannot read properties of undefined
```
(will listen in a different port so nothing wrong, just easy to wrongly rely on that "sync" property)

Maybe this?
```js
address () {
  return this.opened ? this.server.address() : null
}
```

Reply:
not your problem, if users dont follow the contract of the module, undefined behaviour

Reply:
Can't we define it? if code above fixes it

Reply:
server.address already throws if not bound, no need for any additional complexity

Reply:
and/or returns null, can't remember - point being, wait for ready. readme says that pretty clearly